### ISSUE TITLE
The pytorch compilation tool does not support hardsigmoid independent compilation？

### ISSUE THREAD

**Issue by jia0511** [Xilinx/Vitis-AI#1113]
Hi xilinx,

I replaced the sigmoid operator in efficentNet, but the pytorch version compilation tool prompts that the operator is not supported to be compiled in the DPU, which has greatly affected the model performance, but the tf version compilation tool can support the sigmoid operator independently Compile, I would like to know whether independent compilation is supported in vitis-ai3.0 version?

**Comment by huisunCompiler** [comment]
To my knowledge, fix1 + hard-sigmoid + mul + fix2 will be converted to hard-sigmoid-fix op and deployed to DPU if 

- the fix_point of fix2 should be < 7,
- const value = (float) 1.0001220703125

![image](https://user-images.githubusercontent.com/111118020/212305235-82261f86-8af6-4382-bd32-034acf742b7d.png)
![image](https://user-images.githubusercontent.com/111118020/212305745-c98e11cd-b6ee-4ee4-9039-96f268fba566.png)

I don't know which EfficientNet model you use. We only release EfficientNet-lite_tf2 in vitis-AI 3.0 and your model architecture seems to be EfficientNet-b4 or EfficientNet-b7. Could you please provide the arch of DPU and the xmodel so that we can reproduce the problem.

**Comment by cabreraam** [comment]
How are y'all viewing the DPU subgraphs? What is the command? What is this graphical interface?

**Comment by jia0511** [comment]
@huisunCompiler 
I don’t know exactly what fix1 and fix2 are. Is there any pytorch example where fix1 + hard-sigmoid + mul + fix2 is transferred to the hard-sigmoid-fix operator? In addition, for example, replacing sigmoid with SE module hardsigmoid in the efficientnet network will affect the final Accuracy?

**Comment by jia0511** [comment]
@cabreraam 

I put it directly into xmodel through https://netron.app/ website;

**Comment by huisunCompiler** [comment]
@cabreraam 
My netron is a modified version of very old netron. I recommend using the official version https://github.com/lutzroeder/netron, or the website version https://netron.app/.

**Comment by jia0511** [comment]
@cabreraam 

Can you answer my concerns problem？

**Comment by huisunCompiler** [comment]
@jia0511 ,
Sigmoid can't be transferred to hard-sigmoid. Only hard-sigmoid-fix can be deployed to DPU. Here gives an example of `fix1 + hard-sigmoid + mul + fix2 -> hard_sigmoid-fix` and launch on DPU.

```python
import numpy as np
import xir
import math
np.random.seed(2)
data1 = np.random.randint(low=-8, high=8, size=(1, 14, 14, 240))
const = np.array([1.0001220703125], dtype=np.float32)
g = xir.Graph("h")
data = g.create_op(
    "data1", "data", {"data_type": "FLOAT32", "shape": data1.shape})
input1_fix = g.create_op("input1", "fix", {"bit_width": 8, "fix_point": 2,
                                   "if_signed": True, "round_mode": "DPU_ROUND"}, {"input": [data]})
factor = g.create_const_op("factor", const.astype(np.float32))
hardsigmoid = g.create_op("hardsigmoid", "hard-sigmoid", {}, {"input": [input1_fix]})
mul = g.create_op("mul", "mul", {}, {"input": [hardsigmoid, factor]})
hardsigmoid_fix = g.create_op("hardsigmoid_fix", "fix", {"bit_width": 8, "fix_point": 7,
                                   "if_signed": True, "round_mode": "DPU_ROUND"}, {"input": [mul]})
g.serialize("hard_sigmoid.xmodel")
```
Then, use `vai_c_xir` tool to compile "hard_sigmoid.xmodel", you may find the hard_sigmoid are transfered to hard_sigmoid-fix and deployed on DPU. If it is not the case, could you please provide your lib_info, arch, and xmodel to us? The lib_info can be viewed from

`xir graph your_model_after_compilation.xmodel`

> @huisunCompiler I don’t know exactly what fix1 and fix2 are. Is there any pytorch example where fix1 + hard-sigmoid + mul + fix2 is transferred to the hard-sigmoid-fix operator? In addition, for example, replacing sigmoid with SE module hardsigmoid in the efficientnet network will affect the final Accuracy?

**Comment by jia0511** [comment]
My DPU type is vck5000 8PE

**Comment by roycezjq** [comment]
@jia0511  hi, I'm new to this. How did you replace sigmoid with hard sigmoid? Did you replace in the model definition using torch.nn.HardSigmoid, or during the quantization or optimization?
Thanks

**Comment by jia0511** [comment]
Thanks for your reply. I tried to use efficeintNet-b0 model to replace swish with operators, but hard-sigmoid was allocated to CPU during compilation. What is the cause？
[err.log](https://githu
![微信图片_20230407155131](https://user-images.githubusercontent.com/24822418/230566892-8c2ff404-a4bf-48e3-a030-5c76fde25427.png)
b.com/Xilinx/Vitis-AI/files/11177102/err.log)
